### PR TITLE
story/VSPC 120

### PR DESCRIPTION
**Pull request by prjadhav14**
Migrating virtual spaces UI to thymeleaf

### CODE REVIEW

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/add.html
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/add.html:8]: have you made sure that the script is loaded into the final page?

**Review by jdamerow — CHANGES_REQUESTED** [review]
at least some of the html files are also not using Unix line endings, please fix.

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/add.html
Comment (prjadhav14) [reply vspace/src/main/webapp/WEB-INF/views/staff/modules/sequences/add.html:8]: Yes, the script is getting loaded and functionality of adding slides and removing slides by double clicking on the slide name is also working fine.

File: .project
Comment (jdamerow) [reply .project:11]: ?

File: vspace/pom.xml
Comment (jdamerow) [reply vspace/pom.xml]: ?

**Review by jdamerow — CHANGES_REQUESTED** [review]
It doesn't look like the previous comments have been worked on?

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/module.html
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/module.html:41]: let's remove the log statements

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/module.html
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/module.html:63]: remove log

**Review by jdamerow — CHANGES_REQUESTED** [review]
Let's also remove root-context.xml and all the old jsp pages.

File: vspace/src/main/webapp/WEB-INF/spring/root-context.xml
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/spring/root-context.xml:41]: all these beans need to be put into the javaconfig

Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/spring/root-context.xml:41]: or at least the messagesource one if you have tested everything and the default config file is actually not used.

File: vspace/src/main/webapp/WEB-INF/views/accessdenied.html
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/accessdenied.html:10]: this file has windows line ending and I assume it's not the only one.

Comment (jdamerow) [reply vspace/src/main/webapp/WEB-INF/views/accessdenied.html:10]: they all need to be checked. let someone else help you with that if you want.

**Review by jdamerow — CHANGES_REQUESTED** [review]

**Review by jdamerow — CHANGES_REQUESTED** [review]
All the old jsp pages should be deleted.

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
A couple of things:
- the login button should be smaller (maybe let's just use an icon again) and the top nav bar should be less high. The focus of the page is on the spaces, logging in is only important for people who curate the exhibition at the moment.
- the footer text should be smaller and centered
- if there are no modules on the page there shouldn't be a popup
image

- I believe the module list had icons before?
- the login fields look like they are disabled:
image
- signup and forget password links need to look like links (anotehr color or underlined)
- sign up link requires authentication
- after logging in there is no link to the dashboard
- I'm pretty sure in the old version, once logged in, the user would see an edit icon for each space. that's gone.

once the link to the staff site is back, I'll check the rest.

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
- module links are generated incorrectly. It seems to hardcode `vspace`.
- these should be the same as before (icons)
image
- there is something wrong with these icons (grey at bottom) and the text should be aligned with the icons
image
- there needs to be space between icons and text; there shouldn't be list bullets:
image
- there needs to be padding on the right
image
- I just get an empty page when I go here:
https://diging-dev.asu.edu/vspace-dev/staff/space/SPA000000012
- The whole row should be a link or the space/module name should be clearly a link (underlined or different color):
image
- through out the staff side there is padding missing (e.g. sequence page) and there are blue buttons and links throughout (which doesn't fit at all with the color scheme)

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by prjadhav14** [comment]
Reopening to test fix

**Comment by prjadhav14** [comment]
Reopening to check an issue

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
- info icon on public space page should only be shown if there is a description to be shown
- the link editing form is way to spacious and takes up too much space. it should be a lot smaller like before.
image
- creating of a new module link doesn't work
- rotation of icons has some issues (the `i` is not rotated 90 degree, it seems to get fixed after setting some other values though):
image
- why is there a type dropdown for module links? (it's only giving "Module" as an option)
image

- what's up with the buttons on the change image popup (they are also too big)
image

- please check all functionality before reopening the next pr (have someone else help you if needed)

**Comment by prjadhav14** [comment]
Merged with VSPC-138 and some functionality fixes

**Comment by jdamerow** [comment]
- there is something off with the menu (it opens below the space name)
image
- there should be no border around the logout button
image
- these should be icons and have the link color:
image
- the spaces and links need to be closer together, so we can fit more on here:
image
- text in input forms looks like it's disabled (because it's light grey)
- there is still a logout button on the staff page (not an icon)
- remove search bar from staff side
- on module page the following error is thrown: `Uncaught Error: Bootstrap's JavaScript requires jQuery version 1.9.1 or higher, but lower than version 3`
- module page get all messed up with a smaller browser window
image
- after clicking on the folder drop down icon next to a sequence on the module page once, the page stops responding
- what does the info icon do? and the tooltip is not rendered properly:
image
- the icons on the space list that indicate if a space is linked from anywhere are missing and the ones that indicate if a space is published are missing to (I think that was the second indicator, right?)

**Comment by jdamerow** [comment]
Also, I just noticed this:
image

**Comment by jdamerow** [comment]
And remove the filter icon that doesn't seem to do anything.

**Comment by prjadhav14** [comment]
>     * there is something off with the menu (it opens below the space name)
>       image
> 
>     * there should be no border around the logout button
>       image
> 
>     * these should be icons and have the link color:
>       image
> 
>     * the spaces and links need to be closer together, so we can fit more on here:
>       image
> 
>     * text in input forms looks like it's disabled (because it's light grey)
> 
>     * there is still a logout button on the staff page (not an icon)
> 
>     * remove search bar from staff side
> 
>     * on module page the following error is thrown: `Uncaught Error: Bootstrap's JavaScript requires jQuery version 1.9.1 or higher, but lower than version 3`
> 
>     * module page get all messed up with a smaller browser window
>       image
> 
>     * after clicking on the folder drop down icon next to a sequence on the module page once, the page stops responding
> 
>     * what does the info icon do? and the tooltip is not rendered properly:
>       image

The info icon is loaded as in the screenshot. What are the changes expected?
image

> 
>     * the icons on the space list that indicate if a space is linked from anywhere are missing and the ones that indicate if a space is published are missing to (I think that was the second indicator, right?)

**Comment by jdamerow** [comment]
I believe it's the info icon on the module page (title)

**Comment by prjadhav14** [comment]
> I believe it's the info icon on the module page (title)

Oops sorry, my bad.

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
There is no button to create a new slide in a module?

**Comment by jdamerow** [comment]
- The links on the public page to modules dont' seem to work anymore. looks like the 'vspace' url part is hard-coded.
- I can't edit texts or images on slides.

**Comment by jdamerow** [comment]
- After adding a space link I see this:
image
- The following should say: "Indicates links to unpublished spaces"
image
- the question marks need to be whatever icon they should be and it should say the name of the space not the id
image
- I can't delete any slides; nothing happens when I click on the delete icon next to a slide
- this link goes to an non-existing link:
image

**Comment by prjadhav14** [comment]
>     * his link goes to an non-existing link:

**Comment by jdamerow** [comment]
- when creating a new space, I can't choose an existing image
image
- the list of spaces (when you click on the list icon on the staff spaces page) is missing the icons that indicate unpublished or unreachable spaces.
- on slides, images are blown up to a bigger size than they actually are
image
- why does the top right of a slide always say "Branching Point"?
image
- when I edit a slide the form is empty instead of showing the existing values.

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
- what does this link mean? the branching point isn't part of that sequence.
image
- when I add a new branching point block, I get an empty block. I need to reload to see the values.
image
- when adding/editing(?) a text block and the text is rather long, the pop up gets too big for the window and it can't be scrolled down
image

**Comment by prjadhav14** [comment]
>     * what does this link mean? the branching point isn't part of that sequence.
>       image
> 
>     * when I add a new branching point block, I get an empty block. I need to reload to see the values.
>       image
> 
>     * when adding/editing(?) a text block and the text is rather long, the pop up gets too big for the window and it can't be scrolled down
>       image

In the Branching point drop-down we are showing all the choices available. Should we name it as Branching Choices like we had in old layout?

**Comment by jdamerow** [comment]
Hm, not sure how it looked in the old layout, but maybe name it "This Branching Point links to the following Sequences" or something like that, so it's clear what that is?

**Comment by jdamerow** [comment]
Please delete all the jsps and then let's just merge it!

File: vspace/.classpath
Comment (jdamerow) [inline vspace/.classpath:1]: please revert the change to the line endings

**Review by jdamerow — CHANGES_REQUESTED** [review]

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
I was about to merge this but then noticed that for spaces, when I change the window size, the image moves but the links don't. If you are almost done with the space story, then maybe we'll wait till that's done before we merge.
image
image

**Comment by jdamerow** [comment]
Also, I just noticed that when saving exhibition configurations the success message is no longer shown.

File: vspace/src/main/java/edu/asu/diging/vspace/web/staff/DashboardController.java
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/staff/DashboardController.java:33]: wrong line endings

**Review by jdamerow — CHANGES_REQUESTED** [review]

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
the alert box is covering the page heading now and it can't be close and doesn't close on it's own. Ideally it would not cover anything and it should have a close icon.
image

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
It's half cutoff now.
image

**Comment by prjadhav14** [comment]
Reopening to check style changes

**Comment by prjadhav14** [comment]
Reopening to check style changes

**Comment by jdamerow** [comment]
The alert box is still cut off. Just make it on top over everything and make it fade out more slowly.

**Comment by jdamerow** [comment]
There is an issue now with the space link positioning (the links don't move), but let's address that with the other ticket that works on the link positioning.

**Comment by jdamerow** [comment]
🎉

**Pull request merged**
